### ISSUE TITLE
How to use Awkward Arrays on GPUs

### ISSUE THREAD

**Issue by ramankhurana** [scikit-hep/awkward#1139]
### Which documentation?

Python docstrings

### What needs to be documented?

I am wondering if I can use the awkward arrays on GPUs similar to dataframes or numpy arrays, (provided by rapids.ai and other tools). I came across this link
https://awkward-array.org/how-to-math-gpu.html

but it seems to be not in priority list. I just wanted to check if an implementation already exists. In case yes, an example will be really useful. 

Thanks.

**Comment by jpivarski** [comment]
This is understandably confusing: we did some work last summer and had some GPU demonstrators, but GPUs are not ready for use in analysis. The documentation stub is prominent because when this feature _is_ available, it should be highly featured.

The detailed status is that the demonstrators have shown that it's possible with the current architecture, but we wouldn't be able to remove synchronization points between kernel calls that _might_ kill performance if we continue the way we've been headed. Because of this and difficulties extending Awkward Array with JAX and Dask, we started an internal refactoring that replaces a lot of C++ objects with Python objects. (Ironically, that would help us fix the above-mentioned performance issue because we'll be able to delay evaluation of the kernels in such a way that the parallel process won't be interleaved with synchronization points.) We're in the middle of that refactoring.

GPUs are definitely in the plans: [OAC-2103945](https://www.nsf.gov/awardsearch/showAward?AWD_ID=2103945) has complete GPU support as one of its major goals, though that is a 3 year project.
